flow/activities/flowable.go

```diff
@@ -1063,3 +1067,69 @@ func (a *FlowableActivity) RemoveFlowEntryFromCatalog(ctx context.Context, flowN
 
 	return nil
 }
+
+func (a *FlowableActivity) HandleCancelWorkFlow(ctx context.Context, workflowID string) error {
+	errChan := make(chan error, 1)
+	waitForCancelDuration := 1 * time.Minute
+
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, waitForCancelDuration)
+	defer cancel()
+
+	go func() {
+		err := a.TemporalClient.CancelWorkflow(ctxWithTimeout, workflowID, "")
+		errChan <- err
+	}()
+
+	select {
+	case err := <-errChan:
+		if err != nil {
+			slog.Error(fmt.Sprintf("unable to cancel workflow: %s. Attempting to terminate.", err.Error()))
+			terminationReason := fmt.Sprintf("error encountered while trying to cancel workflow %s", workflowID)
+			if err := a.TemporalClient.TerminateWorkflow(ctx, workflowID, "", terminationReason); err != nil {
```

Comment:
Termination seems like the wrong solution. If cancel isn't working it's because the workflow is stuck, possibly waiting on activity to close. Termination won't change that, it'll just orphan these stuck goroutines while we should be fixing them

Reply:
+1, we should only cancel. In case of termination issue, just raise an error because termination = manual intervention.

Also, you don't need a channel I believe now, as the context.WithTimeout should take care of non-await of the call, you can just do a select on `ctx.Done`?

Reply:
Also, need to skip in case PeerFlow was already canceled, does temporal client complain then?

Reply:
I'll double check the behaviour, but https://github.com/temporalio/temporal/issues/2860 suggests cancelling a cancelled workflow doesn't error